pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -194,12 +194,31 @@ type OpenstackClusterProviderSpec struct {
 	// network, a subnet with NodeCIDR, and a router connected to this subnet.
 	// If you leave this empty, no network will be created.
 	NodeCIDR string `json:"nodeCidr,omitempty"`
+
 	// DNSNameservers is the list of nameservers for OpenStack Subnet being created.
 	DNSNameservers []string `json:"dnsNameservers,omitempty"`
+
 	// ExternalNetworkID is the ID of an external OpenStack Network. This is necessary
 	// to get public internet to the VMs.
 	ExternalNetworkID string `json:"externalNetworkId,omitempty"`
 
+	// ManagedAPIServerLoadBalancer defines whether a LoadBalancer for the
+	// APIServer should be created. If set to true the following properties are
+	// mandatory: APIServerLoadBalancerFloatingIP, APIServerLoadBalancerPort
```

Comment:
Why is floating ip field necessary when this field is true if the comment for that field indicates it will be created if not specified?

Reply:
I'm not sure if I get it :). If this field is true that means an APIServerLoadBalancer should be created. If an APIServerLoadBalancer should be created we need a FloatingIP to associate it with the LB.

Reply:
The comment of the floating ip field doesnt't say it will be created if not specified. It says the FloatingIP will be created (in OpenStack) if it doesn't already exist before (in OpenStack). But you have to set the value.

Reply:
Oh I understand, the field is always necessary then right? Thanks for clarifying

Reply:
APIServerLoadBalancerFloatingIP is necessary if ManagedAPIServerLoadBalancer is set to true.